app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java

```diff
@@ -107,4 +150,121 @@ public void prepareView() {
             }
         } //TODO log exception to crashlitycs on else.
     }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if (requestCode == PLACE_REQUEST_ID && resultCode == Activity.RESULT_OK) {
+            Place place = PlacePicker.getPlace(this, data);
+            onNewPlace(place);
+        } else if (requestCode == GPS_REQUEST_ID && resultCode == Activity.RESULT_OK) {
+            startPlaceIntent();
+        }
+    }
+
+    private void onNewPlace(Place place) {
+        tag.setValue(String.valueOf(place.getLatLng().latitude));
+        tag.setSecondValue(String.valueOf(place.getLatLng().longitude));
+        editViewModel.onEditEnded(tag);
+    }
+
+    private void openEditDialog(Tag tag) {
+        EditDialog dialog = EditDialog.newInstance(EditActivity.this, tag);
+        dialog.setViewModel(editViewModel);
+        dialog.show(getFragmentManager(), EditDialog.FRAGMENT_TAG);
+    }
+
+    private void startPlaceIntent() {
+        if (checkGpsEnabled(this)) {
+            openPlacePicker();
+        } else {
+            createLocationRequest(locationSettingsResult -> {
+                final Status status = locationSettingsResult.getStatus();
+                switch (status.getStatusCode()) {
+                    case LocationSettingsStatusCodes.SUCCESS:
+                        openPlacePicker();
+                        break;
+                    case LocationSettingsStatusCodes.RESOLUTION_REQUIRED:
+                        try {
+                            status.startResolutionForResult(
+                                    EditActivity.this,
+                                    GPS_REQUEST_ID);
+                        } catch (IntentSender.SendIntentException e) {
+                            //TODO log crashlytics error
+                        }
+                        break;
+                    default:
+                        break;
+                }
+            });
+        }
+    }
+
+    public boolean checkGpsEnabled(Context context) {
+        LocationManager locationManager = (LocationManager) context
+                .getSystemService(Context.LOCATION_SERVICE);
+        return locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
+    }
+
+    private void createLocationRequest(ResultCallback<LocationSettingsResult> callback) {
+        if (mGoogleApiClient != null && mGoogleApiClient.isConnected()) {
+            openPlacePicker();
+        } else {
+            if (mGoogleApiClient == null) {
+                mGoogleApiClient = new GoogleApiClient.Builder(this)
+                        .addApi(LocationServices.API)
+                        .addConnectionCallbacks(new GoogleApiClient.ConnectionCallbacks() {
+                            @Override
+                            public void onConnected(@android.support.annotation.Nullable Bundle bundle) {
```

Comment:
please use Nullable annotation from javax - in case of passing null value in unit test, findbugs will return error.